plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/widget/GraphWidget.java

```diff
@@ -191,54 +191,21 @@ public Number fromString(String string) {
       }
     });
 
-    xAxis.lowerBoundProperty().bind(xAxis.upperBoundProperty().subtract(visibleTime.multiply(1e3)));
-
-    // Make sure data gets re-added to the chart
-    visibleTime.addListener((__, prev, cur) -> {
-      if (cur.doubleValue() > prev.doubleValue()) {
-        // insert data at the beginning of each series
-        realData.forEach((series, dataList) -> {
-          List<Data<Number, Number>> toAdd = new ArrayList<>();
-          for (int i = 0; i < dataList.getXValues().size(); i++) {
-            double x = dataList.getXValues().get(i);
-            if (x >= xAxis.getLowerBound()) {
-              if (x < series.getData().get(0).getXValue().doubleValue()) {
-                Data<Number, Number> data = dataList.asData(i);
-                if (!toAdd.isEmpty()) {
-                  Data<Number, Number> squarifier = new Data<>(
-                      data.getXValue().doubleValue() - 1,
-                      toAdd.get(toAdd.size() - 1).getYValue().doubleValue()
-                  );
-                  toAdd.add(squarifier);
-                }
-                toAdd.add(data);
-              } else {
-                break;
-              }
-            }
-          }
-          series.getData().addAll(0, toAdd);
-        });
-      }
-    });
-
-    ActionList.registerSupplier(root, () -> {
-      return ActionList.withName(getTitle())
-          .addAction("Clear", () -> {
-            synchronized (queueLock) {
-              chart.getData().forEach(s -> s.getData().clear());
-              queuedData.forEach((s, q) -> q.clear());
-              realData.forEach((s, d) -> d.clear());
-            }
-          });
-    });
+    ActionList.registerSupplier(root, () -> ActionList.withName(getTitle())
+        .addAction("Clear",
+            () -> chart.getDatasets().forEach(s -> {
```

Comment:
Weird formatting. If you pull the "clear" functionality into its own method, this could look something like:


```java
ActionList.registerSupplier(
  root,
  () -> ActionList.withName(getTitle()).withAction("Clear", this::clear)
);
```